### PR TITLE
NAS-123519 / 24.10 / Add globally unique system ID

### PR DESCRIPTION
**Pull request by Qubad786**
### Context
We needed a way to uniquely identify ours systems with some IDs generated and store one time in the configuration database

### Solution
Added a new table in database which holds uuid4 value saved in it on initial install of the system or upgrade of the system.

### CODE REVIEW

File: src/middlewared/middlewared/migration/0009_system_global_id.py
Comment (yocalebo) [inline src/middlewared/middlewared/migration/0009_system_global_id.py:5]: Is this guaranteed this will only ever be called once for the lifetime of the install? Either way, we should probably check if there is a UUID before inserting here always.

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Comment by themylogin** [comment]
Do we need something more complex to prevent issues like people getting duplicate IDs when importing existing config on a different machine?

**Comment by Qubad786** [comment]
> Do we need something more complex to prevent issues like people getting duplicate IDs when importing existing config on a different machine?

@themylogin if an existing config is imported on a different machine, that machine is technically the same now or is that not the case?

**Comment by themylogin** [comment]
> > Do we need something more complex to prevent issues like people getting duplicate IDs when importing existing config on a different machine?
> 
> @themylogin if an existing config is imported on a different machine, that machine is technically the same now or is that not the case?

@yocalebo your feedback is needed here

**Comment by yocalebo** [comment]
> > > Do we need something more complex to prevent issues like people getting duplicate IDs when importing existing config on a different machine?
> > 
> > 
> > @themylogin if an existing config is imported on a different machine, that machine is technically the same now or is that not the case?
> 
> @yocalebo your feedback is needed here

That's correct, this id is _not_ hardware dependent so this is the intended behavior.

File: src/middlewared/middlewared/migration/0009_system_global_id.py
Comment (yocalebo) [reply src/middlewared/middlewared/migration/0009_system_global_id.py:5]: okay

**Review by yocalebo — APPROVED** [review]
Please wait for @themylogin final approval before merge, thanks.

**Pull request merged**
